optashift-employee-rostering-gwtui/src/main/java/org/optaplanner/openshift/employeerostering/gwtui/client/common/Observable.java

```diff
@@ -0,0 +1,27 @@
+package org.optaplanner.openshift.employeerostering.gwtui.client.common;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import org.optaplanner.openshift.employeerostering.gwtui.client.interfaces.Updatable;
+
+public class Observable<T> {
```

Comment:
On one hand, I'd expect errai to have something like this already
On the other hand, I wonder if the abstraction is worth it (it's not like `listeners.forEach((l) -> l.onUpdate(newValue))` is a lot of code).

@tiagobento / @mbarkley wdyt?

Reply:
In fact, `java.util` has this exactly, but for some reason it not in GWT Java emulation libary. Might be able to simulate this with events (`@Observes Event<X> e` and `@Inject Event<X>`

Reply:
Did a refactor to use ErraiMessage bus (a weakly typed equivalent, register SpotListProvider and the like as "Services" and have subscriber subscribe to their events)